iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -98,9 +98,13 @@ void PortManager::doDiscovery()
 void PortManager::handleSenderPorts()
 {
     // get the changes of sender port offer state
-    for (auto senderPortData : m_portPool->senderPortDataList())
+
+    auto& senderList = m_portPool->senderPortDataList();
+    for (auto iter = senderList.begin(); iter != senderList.cend();)
     {
-        SenderPortType senderPort(senderPortData);
+        auto checkIter = iter;
```

Comment:
I think it would be better to write
```
SenderPortData * dataPtr = *iter;
```
Then we have better readable code in the whole function